File: DynamicContentSample/Objective-C/DynamicContentSample/DMCDetailViewController.m
//
//  DMCDetailViewController.m
//  DynamicContentSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

#import "DMCDetailViewController.h"

@interface DMCDetailViewController ()

@end

@implementation DMCDetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    _lblName.text = _result.name_L;
    _lblAddress.text = _result.address_L;
    _lblDetail.text = _result.detail_L;
    _lblTelNo.text = _result.telNo;
    _lblAddInfo.text = _result.addInfo_L;
    _lblWebsite.text = _result.website;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
    if ([segue.identifier isEqualToString:@"detailtoMapSegue"]) {
        DMCMapViewController *mapViewController = segue.destinationViewController;
        mapViewController.result = _result;
    }
}

- (IBAction)btnShare_Clicked:(id)sender {
    NTLocation *location = [[NTLocation alloc] initWithName:_result.name_L
                                                        lat:_result.lat
                                                        lon:_result.lon];
    NSError *error = nil;
    NTShortLinkParameter *param = [[NTShortLinkParameter alloc] initWithStops:@[location]];
    
    NTShortLinkResult *shareResult = [NTShortLinkService execute:param error:&error];
    
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Share" message:shareResult.result preferredStyle:UIAlertControllerStyleAlert];
    [alertController addAction:[UIAlertAction actionWithTitle:@"Copy"
                                                        style:UIAlertActionStyleDefault
                                                      handler:^(UIAlertAction * _Nonnull action) {
                                                          [[UIPasteboard generalPasteboard] setString:shareResult.result];
                                                      }]];
    
    [alertController addAction:[UIAlertAction actionWithTitle:@"Cancel"
                                                        style:UIAlertActionStyleCancel
                                                      handler:nil]];
    
    [self presentViewController:alertController animated:true completion:nil];

}

@end

File: DynamicContentSample/Swift/DynamicContentSample/DMCDetailViewController.swift
//
//  DMCDetailViewController.swift
//  DynamicContentSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

import UIKit
import NOSTRASDK
class DMCDetailViewController: UIViewController {

    @IBOutlet weak var lblName: UILabel!
    @IBOutlet weak var lblAddress: UILabel!
    @IBOutlet weak var lblAdd_Info: UILabel!
    @IBOutlet weak var lblDetail: UILabel!
    @IBOutlet weak var lblTelNo: UILabel!
    @IBOutlet weak var lblWebsite: UILabel!
    
    var result: NTDynamicContentResult! = nil;
    
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        
        lblName.text = result.name_L;
        lblAddress.text = result.address_L;
        lblAdd_Info.text = result.addInfo_L;
        lblDetail.text = result.detail_L;
        lblTelNo.text = result.telNo;
        lblWebsite.text = result.website;
    }

   
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    @IBAction func btnShare_Clicked(_ sender: AnyObject) {
        
        do {
            let location = NTLocation(name: result.name_L, lat: result.lat, lon: result.lon);
            let param = NTShortLinkParameter(stops: [location]);
            let shareResult = try NTShortLinkService.execute(param);
            
            let alertController = UIAlertController(title: "Share", message: shareResult.result, preferredStyle: .alert);
            let actionCopy = UIAlertAction(title: "Copy", style: .default, handler: { (action) in
                UIPasteboard.general.string = shareResult.result;
            })
            let actionCancel = UIAlertAction(title: "Cancel", style: .cancel, handler: nil);
            
            alertController.addAction(actionCopy);
            alertController.addAction(actionCancel);
            
            self.present(alertController, animated: true, completion: nil);
        }
        catch {}
        
        
    }



    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
        if segue.identifier == "detailtoMapSegue" {
            let mapViewController = segue.destination as! DMCMapViewController;
            mapViewController.result = result;
        }
    }


}
